packages/cli/src/handlers/oauthLogin.ts

```diff
@@ -0,0 +1,374 @@
+    // Generate PKCE values
+    const codeVerifier = generateCodeVerifier();
+    const codeChallenge = generateCodeChallenge(codeVerifier);
+    const state = crypto.randomBytes(16).toString('hex');
+
+    // Find an available port
+    const port = await findAvailablePort(8100, 8110);
+    const redirectUri = `http://localhost:${port}/callback`;
+
+    // Create a promise that will be resolved when we get the authorization code
+    let resolveAuth: (value: { code: string; state: string }) => void;
+    let rejectAuth: (reason: Error) => void;
+    const authPromise = new Promise<{ code: string; state: string }>(
+        (resolve, reject) => {
+            resolveAuth = resolve;
+            rejectAuth = reject;
+        },
+    );
+
+    // Create HTTP server to handle the callback
+    const server = http.createServer((req, res) => {
+        if (req.url?.startsWith('/callback')) {
+            const callbackUrl = new URL(req.url, `http://localhost:${port}`);
+            const code = callbackUrl.searchParams.get('code');
+            const returnedState = callbackUrl.searchParams.get('state');
+            const error = callbackUrl.searchParams.get('error');
+
+            if (error) {
+                rejectAuth(new AuthorizationError(`OAuth error: ${error}`));
+                res.writeHead(400, { 'Content-Type': 'text/html' });
+                res.end(`
+                    <html>
+                        <head>
+                            <style>${oauthPageStyles}</style>
+                        </head>
+                        <body>
+                            <div class="stack">
+                                
+                                <div class="container error">
+                                    <svg class="icon" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+                                        <circle cx="12" cy="12" r="10" stroke="currentColor" stroke-width="2"/>
+                                        <path d="M15 9l-6 6M9 9l6 6" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+                                    </svg>
+                                    <h1>Authentication Failed</h1>
+                                    <p>Error: ${error}</p>
+                                    <p>You can close this window and try again.</p>
+                                </div>
+                            </div>
+                        </body>
+                    </html>
+                `);
+                return;
+            }
+
+            if (!code || !returnedState) {
+                rejectAuth(
+                    new AuthorizationError(
+                        'Missing authorization code or state',
+                    ),
+                );
+                res.writeHead(400, { 'Content-Type': 'text/html' });
+                res.end(`
+                    <html>
```

Comment:
html for successful oauth flow

<img width="551" height="364" alt="Screenshot from 2025-07-18 14-53-57" src="https://github.com/user-attachments/assets/0cb2d747-5215-461b-bac5-687b218765ad" />